### PR TITLE
Improve documentation: general fixes / API linking / add more API docs CB-1150

### PR DESCRIPTION
**Pull request by brickZA**

### CODE REVIEW

File: doc/tutorial.rst [hunk 1]
```diff
@@ -92,7 +92,7 @@ but doesn't want to wait for a reply, the
 Note that the :func:`reply_cb` and :func:`inform_cb` callback functions are both
```
Comment (profcalculus) [inline doc/tutorial.rst]: parens after the function names?

File: doc/tutorial.rst [hunk 2]
```diff
@@ -92,7 +92,7 @@ but doesn't want to wait for a reply, the
...
 called inside the client's event-loop thread so should not perform any
 operations that block. If needed, pass the data out to from the callback
```
Comment (profcalculus) [inline doc/tutorial.rst]: 'data out from'

File: doc/tutorial.rst [hunk 3]
```diff
@@ -118,17 +118,19 @@ server.
...
-    :meth:`is_connected`.
+    :meth:`is_connected() <DeviceClient.is_connected>`.
 
-When the :class:`DeviceClient` thread receives a completed message
-:meth:`handle_message` is called.  The default :meth:`handle_message`
-implementation calls one of :meth:`handle_reply`, :meth:`handle_inform`
-or :meth:`handle_request` depending on the type of message received.
+When the :class:`DeviceClient` thread receives a completed message,
+:meth:`handle_message` is called.  The default :meth:`handle_message()
+<DeviceClient.handle_message>` implementation calls one of :meth:`handle_reply
```
Comment (profcalculus) [inline doc/tutorial.rst]: handle_reply()

File: doc/tutorial.rst [hunk 4]
```diff
@@ -144,7 +146,7 @@ Similarly an inform message named :samp:`bar` will be dispatched to
 :meth:`unhandled_reply`, :meth:`unhandled_inform` or :meth:`unhandled_request`
```
Comment (profcalculus) [inline doc/tutorial.rst]: parens after method names?

File: doc/tutorial.rst [hunk 5]
```diff
@@ -389,35 +391,37 @@ change server behaviour; furthermore version info may need to be passed to the
...
-with "request\_".  These methods take two arguments -- the client socket that
-the request came from and the request message.  Notice that the message argument
-is missing from the methods in the example. This is a result of the
-:meth:`request <katcp.kattypes.request>` decorator that has been applied to the
-methods.
-
-The :meth:`request <katcp.kattypes.request>` decorator takes a list of
-:class:`kattype <katcp.kattypes.KatcpType>` objects describing the request
+with "request\_".  These methods take two arguments -- the client-request object
+(abstracts the client socket and the request context) that the request came from
```
Comment (profcalculus) [inline doc/tutorial.rst]: from,

File: doc/tutorial.rst [hunk 6]
```diff
@@ -474,7 +478,7 @@ behaviour by default, while using :class:`AsyncClient` or
 :class:`AsyncDeviceServer` will by default use `tornado.ioloop.IOLoop.current()`
 as the ioloop (can be overidden using their `set_ioloop` methods), and won't
 enable thread safety by default (can be overridden using
-:meth:`AsyncDeviceServer.set_concurrency_operations` and
+:meth:`AsyncDeviceServer.set_concurrency_options` and
```
Comment (profcalculus) [inline doc/tutorial.rst]: ()

Comment (profcalculus) [inline doc/tutorial.rst]: ()

File: doc/tutorial.rst [hunk 7]
```diff
@@ -474,7 +478,7 @@ behaviour by default, while using :class:`AsyncClient` or
...
 :meth:`AsyncClient.enable_thread_safety`)
```
Comment (profcalculus) [inline doc/tutorial.rst]: ()

File: katcp/server.py [hunk 8]
```diff
@@ -1248,6 +1248,24 @@ def set_ioloop(self, ioloop=None):
...
+
+        thread_safe : bool
+            If True, make the server public methods thread safe. Incurs
+            performance overhead.
+        handler_thread : bool
+            Can only be set if `thread_safe` is True. Handle all requests (even
+            from different clients) in a separate, single, request-handling
+            thread. Blocking request handlers will prevent the server from
+            handling new requests from any client, but sensor strategies should
+            still function. This more or less mimicks the behaviour of a server
```
Comment (profcalculus) [inline katcp/server.py]: s/mimicks/mimics

File: doc/tutorial.rst (hunk 1 above)
Comment (brickZA) [inline doc/tutorial.rst]: Strange foible of sphinx, if you do the :func: thing naked it adds the parens for you, but if you do it as :func:`reply_cb <katcp.blah.reply_cb>` it takes your description literally and doesn't add parens. So in this case it should be fine without the ().

Comment (profcalculus) [inline doc/tutorial.rst]: OK, didn't know that...

File: doc/tutorial.rst (hunk 2 above)
Comment (brickZA) [inline doc/tutorial.rst]: Change should land with next push.

File: doc/tutorial.rst (hunk 3 above)
Comment (brickZA) [inline doc/tutorial.rst]: Change should land with next push.

File: doc/tutorial.rst (hunk 4 above)
Comment (brickZA) [inline doc/tutorial.rst]: Same as for first comment.

File: doc/tutorial.rst (hunk 5 above)
Comment (brickZA) [inline doc/tutorial.rst]: Same as for first comment.

File: doc/tutorial.rst (hunk 6 above)
Comment (brickZA) [inline doc/tutorial.rst]: Same as first comment.

File: doc/tutorial.rst (hunk 7 above)
Comment (brickZA) [inline doc/tutorial.rst]: Same as first comment.

File: katcp/server.py (hunk 8 above)
Comment (brickZA) [inline katcp/server.py]: Change should land with next push.

**Pull request merged**